### PR TITLE
Fix overlay bug when offline

### PR DESCRIPTION
**Pull request by wu-hui**
Fix: https://github.com/firebase/firebase-android-sdk/issues/3528

### CODE REVIEW

**Review by dconeybe — CHANGES_REQUESTED** [review]
This PR needs a change log entry. Also, please apply the feedback from @schmidt-sebastian, if any.

File: firebase-firestore/CHANGELOG.md [hunk 1]
```diff
@@ -2,20 +2,21 @@ Android changes are not released automatically. Ensure that changes are released
...
+- [fixed] Fixed missing document fields issue with offline overlays (#3528)
 
 # 24.0.2
 - [fixed] Fixed an issue of long grpc reconnection period, when App moves to
   foreground after staying in background for a while.
 - [fixed] Fixed an AppCheck issue that caused Firestore listeners to stop
   working and receive a "Permission Denied" error. This issue only occurred for
   AppCheck users that set their expiration time to under an hour.
-- [fixed] Fixed a potential problem during Firestore's shutdown that prevented
+- [fixed] Fixed a potential problem during Firestore's shutdown that prevented
```
Comment (schmidt-sebastian) [inline firebase-firestore/CHANGELOG.md:19]: Please revert.

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java [hunk 2]
```diff
@@ -884,4 +884,39 @@ public void testCollectionGroupQueriesWithWhereFiltersOnArbitraryDocumentIds() {
                 .get());
     assertEquals(asList("cg-doc2"), querySnapshotToIds(querySnapshot));
   }
+
+  // See: https://github.com/firebase/firebase-android-sdk/issues/3528
+  @Test
+  public void testAddThenUpdatesWhileOffline() {
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java:890]: Ideally, these would be LocalStoreTests.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/Mutation.java [hunk 3]
```diff
@@ -93,7 +94,7 @@ public static Mutation calculateOverlayMutation(MutableDocument doc, @Nullable F
     }
 
     // mask == null when there are Set or Delete being applied to get to the current document.
-    if (mask == null) {
+    if (mask == null || doc.getVersion().equals(SnapshotVersion.NONE)) {
```
Comment (schmidt-sebastian) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/Mutation.java:97]: Can you update the comment?

**Review by schmidt-sebastian — APPROVED** [review]
Thanks

File: firebase-firestore/CHANGELOG.md (hunk 1 above)
Comment (wu-hui) [reply firebase-firestore/CHANGELOG.md]: Done.

File: firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java (hunk 2 above)
Comment (wu-hui) [reply firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java:891]: Added TODO.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/Mutation.java (hunk 3 above)
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/Mutation.java]: Stays unchanged now.

**Comment by wu-hui** [comment]
> This PR needs a change log entry. Also, please apply the feedback from @schmidt-sebastian, if any.

Done.

**Review by dconeybe — APPROVED** [review]

**Pull request merged**